src/Application/Processor/ProductLowestPriceBeforeDiscountProcessor.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\PriceHistoryPlugin\Application\Processor;
+
+use Sylius\Component\Channel\Context\ChannelContextInterface;
+use Sylius\PriceHistoryPlugin\Domain\Model\ChannelInterface;
+use Sylius\PriceHistoryPlugin\Domain\Model\ChannelPricingInterface;
+use Sylius\PriceHistoryPlugin\Domain\Model\ChannelPricingLogEntryInterface;
+use Sylius\PriceHistoryPlugin\Domain\Repository\ChannelPricingLogEntryRepositoryInterface;
+use Webmozart\Assert\Assert;
+
+final class ProductLowestPriceBeforeDiscountProcessor implements ProductLowestPriceBeforeDiscountProcessorInterface
+{
+    private const PERIOD_IN_DAYS = 30;
+
+    public function __construct(
+        private ChannelContextInterface $channelContext,
+        private ChannelPricingLogEntryRepositoryInterface $channelPricingLogEntryRepository,
+    ) {
+    }
+
+    public function process(ChannelPricingInterface $channelPricing): void
+    {
+        /** @var ChannelInterface $channel */
+        $channel = $this->channelContext->getChannel();
+
+        if ($channel->isLowestPriceForDiscountedProductsVisible() === false) {
+            $channelPricing->setLowestPriceBeforeDiscount(null);
+
+            return;
+        }
```

Comment:
```suggestion
```
I'd be for always setting it and using the channel's flag to decide only whether to show it or not. Right now it'd take fewer resources when saving a price change, but when enabling the flag on a channel, you'd need to reprocess every price within it.

Reply:
Good point. So for now, I will leave a todo flag here:
https://github.com/Sylius/PriceHistoryPlugin/pull/32/files#diff-f39566ed0bc17aa2aa040f02494e53e0aa9d92052744a37ac50745d760372b55R85